docker/conda_build.dockerfile

```diff
@@ -23,7 +23,7 @@ ENV PATH /opt/conda/bin:$PATH
 
 # install python 3.10 because that's what apt uses
 RUN conda update conda
-RUN conda install "python=3.10"
+RUN conda install "python=3.10.12"
```

Comment:
I'm anxious about this... Any time we pin a version, we fall behind.  Can you try 3.12?

Reply:
Sure, but the apt build currently uses Python 3.10.12.

Reply:
it failed... there must be something in newer python that we need to catch up to

Reply:
The scikit-build-core branch is working with the latest Python and is only waiting for the MOAB release. Alternatively, can we do something similar to what we did with pydagmc?

Reply:
It would be nice to know what the cause of the problem is, though.